src/sentry/static/sentry/app/views/settings/components/forms/projectMapperField.tsx

```diff
@@ -0,0 +1,157 @@
+  private sentryProjectRef = React.createRef<typeof SelectControl>();
+  private mappedRef = React.createRef<typeof SelectControl>();
+
+  renderField = (props: RenderProps) => {
+    const {onChange, onBlur, value, mappedDropdown} = props;
+    const existingValues: Array<[number, string | number]> = value || [];
+    const sentryProjects = props.sentryProjects || [];
+    const mappedDropdownItems = mappedDropdown?.items || [];
+
+    // create maps by the project id for constant time lookups
+    const sentryProjectsById = Object.fromEntries(
+      sentryProjects.map(project => [project.id, project])
+    );
+
+    const mappedItemsByValue = Object.fromEntries(
+      mappedDropdownItems.map(item => [item.value, item])
+    );
+
+    //build sets of values used so we don't let the user select them twice
+    const projectIdsUsed = new Set(existingValues.map(tuple => tuple[0]));
+    const mappedValuesUsed = new Set(existingValues.map(tuple => tuple[1]));
+
+    const projectOptions = sentryProjects
+      .filter(project => !projectIdsUsed.has(project.id))
+      .map(({slug, id}) => ({label: slug, value: id}));
+
+    const mappedItemsToShow = mappedDropdownItems.filter(
+      item => !mappedValuesUsed.has(item.value)
+    );
+
+    const handleAdd = () => {
+      const {value: sentryProjectId} = this.sentryProjectRef.current.state.value;
+      const {value: mappedValue} = this.mappedRef.current.state.value;
+      //add the new value to the list of existing values
+      const projectMappings = [...existingValues, [sentryProjectId, mappedValue]];
+      //trigger events so we save the value and show the check mark
+      onChange?.(projectMappings, []);
+      onBlur?.(projectMappings, []);
+    };
+
+    const renderItem = (itemTuple: [number, any]) => {
+      const [projectId, mappedValue] = itemTuple;
+      const {slug} = sentryProjectsById[projectId];
+      const {label: itemLabel} = mappedItemsByValue[mappedValue];
+      return (
+        <Item key={projectId}>
+          <ItemValue>{slug}</ItemValue> <ItemValue>{itemLabel}</ItemValue>
+        </Item>
+      );
+    };
+
+    const customValueContainer = containerProps => {
+      const valueList = containerProps.getValue();
+      //if no value set, we want to return the default component that is rendered
+      if (valueList.length === 0) {
+        return <components.ValueContainer {...containerProps} />;
+      }
+      const projectId = valueList[0].value;
+      const project = sentryProjectsById[projectId];
+      if (!project) {
+        return <components.ValueContainer {...containerProps} />;
+      }
+      return (
+        <components.ValueContainer {...containerProps}>
+          <IdBadge
+            project={project}
+            avatarSize={20}
+            displayName={project.slug}
+            avatarProps={{consistentWidth: true}}
+          />
+        </components.ValueContainer>
+      );
+    };
+
+    const customOptionProject = projectProps => {
```

Comment:
@billyvg do we already have a component used for project selection? Feels like maybe we should if we don't?

Reply:
We do (e.g. global selection header), but this component has more to it than just selecting a project.

Reply:
@EvanPurkhiser Are you talking about `ContextPickerModal`? If so, I added that :)

Anyways, this component isn't much more than a wrapper around `IdBadge` with a project lookup.